src/main/java/org/example/coffeeshop/service/impl/UserServiceImpl.java

```diff
@@ -15,15 +21,51 @@ public class UserServiceImpl implements UserService {
 
     private final UserRepository userRepository;
     private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
 
     @Override
-    public User save(User user) {
-        user.setPassword(passwordEncoder.encode(user.getPassword()));
-        return userRepository.save(user);
+    public Optional<User> findByEmail(String email) {
+        return userRepository.findByEmail(email);
     }
```

Comment:
Service shouldn't return Optional<User>.
That's should throw exception when user is not present